pangeo_forge/recipe.py

```diff
@@ -420,15 +438,21 @@ class NetCDFtoZarrMultiVarSequentialRecipe(NetCDFtoZarrRecipe):
     """There are muliples sequences of input files (but all along the same dimension.)
     Different variables live in different files.
 
+    :param processed_input_urls: The inputs already used to generate the existing dataset.
     :param input_pattern: An pattern used to generate the input file names.
     """
 
+    processed_input_urls: Iterable[str] = field(repr=False, default_factory=list)
     input_pattern: VariableSequencePattern = field(default_factory=VariableSequencePattern)
 
     def __post_init__(self):
         super().__post_init__()
         self._variables = self.input_pattern.keys["variable"]
-        self._inputs = {k: v for k, v in self.input_pattern}
+        self.processed_input_nb = len(self.processed_input_urls) // len(self._variables)
+        self._inputs = {
+            k: {"url": v, "processed": v in self.processed_input_urls}
```

Comment:
Here the `input_pattern` generates the input file URLs including the already processed ones (which is different than for the `NetCDFtoZarrSequentialRecipe`). This allows to tag the already processed URLs.
If the `input_pattern` didn't generate the already processed URLs, the pattern would have to know about the (number of) already processed URLs, which is not great.